### PR TITLE
Code refactor for `mnist_trainer` example

### PR DESCRIPTION
**Pull request by ankitgola005**
<details>
  <summary><b>Code refactor for mnist_trainer example</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>
Code refactor, decoupling model and plugin initializations.

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: examples/pytorch/mnist_trainer.py [hunk 1]
```diff
@@ -63,54 +64,73 @@ def parse_args():
...
-    if run_type == "recipe_caching":
-        os.environ["PT_HPU_RECIPE_CACHE_CONFIG"] = "tmp/recipes,True,1024"
+@contextmanager
+def set_env_vars(env_dict):
+    """CM to set env."""
+    active = bool(env_dict)  # Set to True if env_dict is not None or empty
+    original_values = {key: os.environ.get(key, None) for key in env_dict} if active else {}
+
+    if active:
+        try:
```
Comment (Borda) [inline examples/pytorch/mnist_trainer.py:74]: if no exception is expected, then just use suppress context

File: examples/pytorch/mnist_trainer.py [hunk 2]
```diff
@@ -23,11 +24,11 @@
 if module_available("lightning"):
     from lightning.pytorch import Trainer, seed_everything
     from lightning.pytorch.demos.mnist_datamodule import MNISTDataModule
-    from lightning.pytorch.plugins.precision import MixedPrecisionPlugin
+    from lightning.pytorch.plugins.precision import MixedPrecision
```
Comment (Borda) [inline examples/pytorch/mnist_trainer.py:27]: Shall we then also bump min dependency?

File: examples/pytorch/mnist_trainer.py [hunk 3]
```diff
@@ -63,54 +64,72 @@ def parse_args():
...
+
+
+def run_trainer(model, data_module, plugin, devices=1, strategy=None):
+    """Run trainer.fit with given parameters."""
     trainer = Trainer(
         accelerator=HPUAccelerator(),
-        devices=_devices,
-        strategy=_strategy,
+        devices=devices,
+        strategy=strategy if strategy is not None else HPUParallelStrategy() if devices > 1 else SingleHPUStrategy(),
```
Comment (Borda) [inline examples/pytorch/mnist_trainer.py:91]:
```suggestion
        strategy=strategy or HPUParallelStrategy() if devices > 1 else SingleHPUStrategy(),
```
I would rather move `HPUParallelStrategy() if devices > 1 else SingleHPUStrategy()` out of the argument to improve redability

**Review by Borda — APPROVED** [review]

File: examples/pytorch/mnist_trainer.py (hunk 2 above)
Comment (jerome-habana) [reply examples/pytorch/mnist_trainer.py:27]: Is it not implicit based on the support matrix per release ?

File: examples/pytorch/mnist_trainer.py (hunk 3 above)
Comment (jerome-habana) [reply examples/pytorch/mnist_trainer.py:91]: +1

**Pull request merged**